File: backend/app/models/product.py
"""Modelo del catálogo de productos de venta (módulo "Productos")."""

import uuid
from datetime import datetime

from sqlalchemy import Boolean, DateTime, ForeignKey, Integer, Numeric, String, Text, func
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import Mapped, mapped_column

from app.db.session import Base
from app.models.base import UUIDPkMixin


class SaleProduct(UUIDPkMixin, Base):
    """Producto de venta retail (croquetas, premios, camas, etc.)."""

    __tablename__ = "sale_products"

    clinic_id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True), ForeignKey("clinics.id"), nullable=False, index=True
    )
    name: Mapped[str] = mapped_column(String(150), nullable=False)
    category: Mapped[str] = mapped_column(String(50), nullable=False, index=True)
    price: Mapped[float | None] = mapped_column(Numeric(10, 2))
    stock_quantity: Mapped[int] = mapped_column(
        Integer, nullable=False, default=0, server_default="0"
    )
    photo_url: Mapped[str | None] = mapped_column(Text)
    active: Mapped[bool] = mapped_column(
        Boolean, nullable=False, default=True, server_default="true"
    )
    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), server_default=func.now(), nullable=False
    )

File: frontend/src/lib/product.ts
export interface SaleProduct {
  id: string
  clinic_id: string
  name: string
  category: string
  price: number | null
  stock_quantity: number
  photo_url: string | null
  active: boolean
  created_at: string
}

export const PRODUCT_CATEGORY_SUGGESTIONS = [
  'Alimento',
  'Premios',
  'Juguetes',
  'Ropa',
  'Camas',
  'Platos y accesorios',
  'Higiene',
  'Farmacia',
  'Otro',
]
